minor-mode.lisp

```diff
@@ -0,0 +1,545 @@
+(in-package :stumpwm)
+
+(export '(define-minor-mode
+          minor-mode-id
+          minor-mode-scope
+          enable-minor-mode
+          disable-minor-mode
+          minor-mode-on-enable
+          minor-mode-on-disable
+          minor-mode-keymap
+          minor-mode-lighter
+          generate-keymap))
+
+(defvar *minor-mode*)
+
+(defvar *active-global-minor-modes* nil
+  "A list of active minor modes with a global scope.")
+
+(defun global-minor-modes (&optional ignore)
+  (declare (ignore ignore))
+  *active-global-minor-modes*)
+
+(defun set-global-minor-modes (ignore value)
+  (declare (ignore ignore))
+  (setf *active-global-minor-modes* value))
+
+(defsetf global-minor-modes set-global-minor-modes)
```

Comment:
For simple cases one can define a setf function directly. Although unless the plan is to add hooks for extension or something similar, wouldn't using the global variable directly be good enough?

```suggestion
(defun (setf global-minor-modes) (value ignore)
  (declare (ignore ignore))
  (setf *active-global-minor-modes* value))
```

Reply:
Accessing the global minor mode list through setter/getter functions was done to ease code generation in macros, see `define-enable-methods`. While the variable can be used directly, it does mean adding complexity to the enable/disable method generation.